File: 30_days_of_code/python/day_12.py
class Student:
    def __init__(self,firstName,lastName,phone):
        self.firstName=firstName
        self.lastName=lastName
        self.phone=phone
    def display(self):
        print ("First Name:",self.firstName)
        print ("Last Name:",self.lastName)
        print ("Phone:",self.phone)

class Grade(Student):
    def __init__(self, firstName, lastName, phone, score):
        self.firstName = firstName
        self.lastName = lastName
        self.phone = phone
        self.score = score
    def calculate(self):
        if(90 <= self.score <= 100):
            return 'O'
        if(75 <= self.score < 90):
            return 'E'
        if(60 <= self.score < 75):
            return 'A'
        if(40 <= self.score < 60):
            return 'B'
        if(self.score < 40):
            return 'D'

firstName=input().strip()
lastName=input().strip()
phone=int(input())
score=int(input())
stu=Grade(firstName,lastName,phone,score)
stu.display()
print ("Grade:",stu.calculate()) 

File: 30_days_of_code/ruby/day_12.rb
class Student
    def initialize(firstName,lastName,phone)
        @firstName=firstName
        @lastName=lastName
        @phone=phone
    end
    def display()
        print("First Name: ",@firstName,"Last Name: ",@lastName+"Phone: ",@phone)
    end
end

class Grade <Student
    def initialize(firstName, lastName, phone, score)
        @firstName = firstName
        @lastName = lastName
        @phone = phone
        @score = score
    end
    def calculate()
        return 'O' if (90 <= @score and @score <= 100)
        return 'E' if (75 <= @score and @score < 90)
        return 'A' if (60 <= @score and @score < 75)
        return 'B' if (40 <= @score and @score < 60)
        return 'D' if (@score < 40)
    end
end

firstName=gets
lastName=gets
phone=gets.to_i
score=gets.to_i
g=Grade.new(firstName,lastName,phone,score)
g.display           
print("\nGrade: ",g.calculate)
